### DISCUSSION TITLE
Unexpected output with --no-optimize cli option

### DISCUSSION THREAD

**Discussion by avalanche1** [parcel-bundler/parcel#5764] [Q&A] [answered]
```ts
import {browser} from "webextension-polyfill-ts";

const foo = async () => console.log(await browser.storage.local.get());
foo();
```
produces this:
```js
  var $d03511d67c54a6de175fe6ab6ad60bdb$export$browser = $91e2c518faad6af4b6659069b488a574$exports;
  $d03511d67c54a6de175fe6ab6ad60bdb$exports.browser = $d03511d67c54a6de175fe6ab6ad60bdb$export$browser;
  const $48bd52ff54c9d5af24a9b6c67fd1cfa0$var$foo = async () => console.log(await $d03511d67c54a6de175fe6ab6ad60bdb$exports.browser.storage.local.get());
  $48bd52ff54c9d5af24a9b6c67fd1cfa0$var$foo();
```
I would expect the `--no-optimize` option to keep my code unchanged.
Why is this strange transformation happening?
Can I get rid of it?

**Comment by devongovett** [comment, marked as answer]
This is scope hoisting. See https://v2.parceljs.org/features/scope-hoisting/#motivation-and-advantages-of-scope-hoisting for a description.

It can be disabled with `--no-scope-hoist`, but this will result in much larger output.

**Reply by avalanche1** [reply]
> It can be disabled with --no-scope-hoist, but this will result in much larger output.

Sometimes that's exactly what is needed 😀

**Comment by danieltroger** [comment]
`--no-optimize` basically just disables minification, it was called `--no-minify` before.

Your code will still be bundled and transpiled if your specified target needs it.

**Reply by georgeemichel** [reply]
--no-minify works for me
